server/controllers/api_controller.go

```diff
@@ -197,7 +197,7 @@ func (a *APIController) apiParseAndValidate(r *http.Request) (*APIRequest, *comm
 	if err != nil {
 		return nil, nil, http.StatusBadRequest, err
 	}
-	cloneURL, err := a.VCSClient.GetCloneURL(VCSHostType, request.Repository)
+	cloneURL, err := a.VCSClient.GetCloneURL(a.Logger, VCSHostType, request.Repository)
```

Comment:
Is there a better pattern to avoid having to pass in the logger in so many functions as a parameter?

Is there a way to globally set the logger and initialize it once?

Reply:
It would need the VCS client objects being created once per command rather than once globally, which would be a huge refactor, so this is the best compromise.